### DISCUSSION TITLE
Getting size in bytes of a zarr group

### DISCUSSION THREAD

**Discussion by aliaksei-chareshneu** [zarr-developers/zarr-python#1668] [Q&A]
Dear all,

Could you tell me please if it is possible to get size of a zarr  group in bytes?

Thank you for any input,
Aliaksei

**Comment by d-v-b** [comment]
do you want the size of the group metadata in bytes (generally small, because it's JSON), or the size of all the group + arrays stored inside that group, in bytes (potentially very large, depending on the number of arrays and how big they are)

**Reply by aliaksei-chareshneu** [reply]
Thank you for the response. The second option - size of all the group + arrays + subgroups - anything that is inside that group, recursively.

**Reply by d-v-b** [reply]
the [`getsize`](https://github.com/zarr-developers/zarr-python/blob/3db41760e18fb0a69b5066e8c7aba9752a8c474e/zarr/storage.py#L251) method defined on the store class (accessed via `group.store`) should do this. But I imagine this could take a long time to run, if you have arrays with many chunks

**Reply by aliaksei-chareshneu** [reply]
Thank you very much. However, it looks like it does not take into acoount multiple nested groups. Is it possible to do in case there are multiple nested groups?
